server/src/main/java/io/crate/execution/dml/Indexer.java

```diff
@@ -536,24 +536,34 @@ public Indexer(String indexName,
      * @param getRef A function that returns a reference for a given column ident based on the current cluster state
      */
     public void updateTargets(Function<ColumnIdent, Reference> getRef) {
-        for (int i = 0; i < columns.size(); i++) {
-            var oldRef = columns.get(i);
+        var it = columns.iterator();
+        var idx = 0;
+        while (it.hasNext()) {
+            var oldRef = it.next();
             if (oldRef.oid() == COLUMN_OID_UNASSIGNED) {
-                // try to resolve the column again, new reference may have been added to the table already or
-                // to the list of new columns returned by the {@link AddColumnTask}
+                // try to resolve the column again, new reference may have been added to or dropped of the table or
+                // the reference may be invalid
                 Reference newRef = getRef.apply(oldRef.column());
```

Comment:
question on code comment: is "or dropped" real reason for removing value indexer? AFAIK, we ignore dropped columns when we build `DocTableInfo` object, `getRef` should never operate with such columns?

I guess actual reason of this change is only invalid column in the targets (`testCopyFromFileWithInvalidColumns`)?

Reply:
I think there can be a race-condition whereas a column was dropped after the indexer processed the columns and create a child value indexer but before `index` is called. On indexing, the stored reference inside `childColumns` is used instead of calling a (stale) getRef.